app/controlplane/internal/usercontext/jwt_middleware.go

```diff
@@ -0,0 +1,231 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package usercontext
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	errorsAPI "github.com/go-kratos/kratos/v2/errors"
+	"github.com/go-kratos/kratos/v2/middleware"
+	"github.com/go-kratos/kratos/v2/transport"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+const (
+	// bearerWord the bearer key word for authorization
+	bearerWord string = "Bearer"
+	// authorizationKey holds the key used to store the JWT Token in the request tokenHeader.
+	authorizationKey string = "Authorization"
+	// reason holds the error reason.
+	reason string = "UNAUTHORIZED"
+	// RobotAccountProviderKey is the key for robot account token provider
+	RobotAccountProviderKey = "robotAccountProvider"
+	// APITokenProviderKey is the key for api token provider
+	APITokenProviderKey = "apiTokenProvider"
+)
+
+var (
+	ErrMissingJwtToken        = errorsAPI.Unauthorized(reason, "JWT token is missing")
+	ErrMissingKeyFunc         = errorsAPI.Unauthorized(reason, "keyFunc is missing")
+	ErrTokenInvalid           = errorsAPI.Unauthorized(reason, "Token is invalid")
+	ErrTokenExpired           = errorsAPI.Unauthorized(reason, "JWT token has expired")
+	ErrTokenParseFail         = errorsAPI.Unauthorized(reason, "Fail to parse JWT token ")
+	ErrUnSupportSigningMethod = errorsAPI.Unauthorized(reason, "Wrong signing method")
+	ErrWrongContext           = errorsAPI.Unauthorized(reason, "Wrong context for middleware")
+	ErrNeedTokenProvider      = errorsAPI.Unauthorized(reason, "Token provider is missing")
+)
+
+type JWTAuthContext struct {
+	Claims         jwt.Claims
+	IsAPIToken     bool
```

Comment:
Why do we need two booleans? What about just the providerKey? which then can be compared to the const you have above?

Reply:
It makes sense, I'll leverage the use of the provider keys and export them to do the validation.